### PR TITLE
Add TargetPlatform linux/windows breaking change migration guide.

### PR DESCRIPTION
**Pull request by gspencergoog**
This adds a migration guide to go with the flutter/flutter#51519 PR.

### CODE REVIEW

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+---
+title: Adding `linux` and `windows` to [`TargetPlatform`][] enum
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: Reemove link from title.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+description: Two new values were added to the [`TargetPlatform`][] enum that could require additional cases in switch statements that switch on a [`TargetPlatform`][].
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: The meta data shouldn't have any links. Remove both of these

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+---
+
+## Summary
+
+Two new values were added to the [`TargetPlatform`][] enum that could require additional cases in switch statements that switch on a [`TargetPlatform`][] and don't include a `default:` case.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: It's sufficient to link on the first instance, or much later in the doc. Remove the second link (but keep the codefonted text).

Also, line length should be less than 80 chars and this link is much longer.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+```dart
+enum TargetPlatform {
+  android,
+  fuchsia,
+  iOS,
+  macOS,
+}
+```
+
+A `switch` statement only needed to handle these cases, and desktop applications which wanted to run on Linux or Windows usually had a shim something like this in their main:
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: Line is too long.

Also which => that

in their main => in their `main()` method:

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  android,
+  fuchsia,
+  iOS,
+  linux, // new value
+  macOS,
+  windows, // new value
+}
+```
+
+And the shim in `main()` is no longer required.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: I don't know what you mean by "shim" but will other devs know?

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+This can cause the Dart analyzer to give the [`missing_enum_constant_in_switch`][]
+warning for switch statements which don't include a `default` case. Writing a
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: which => that

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+switch without a `default:` case is the recommended way to handle enums, since
+the analyzer can then help you find any cases which aren't handled.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: which => that

Guidance on what to use when:
https://www.quickanddirtytips.com/education/grammar/which-versus-that-0

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+## Migration guide
+
+In order to migrate to the new enum, and avoid the analyzer's [`missing_enum_constant_in_switch`][] error, which looks like:
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: This "which" is fine!

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+Having `default:` cases in such switch statements isn't recommended, because
+then the analyzer can't help you find all the cases which need to be handled.
+
+Also, any shims like the one above that set the
+[`debugDefaultTargetPlatformOverride`][] are no longer needed for Linux and Windows
+applications.
+
+## Timeline
+
+This change was made in March 2020.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: Do you have the release number? We can't merge this PR until we have an actual release #.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+API documentation:
+* [`TargetPlatform`][]
+
+Relevant issues:
+* [Issue #31366][]
+
+Relevant PRs:
+* [Add Windows, and Linux as TargetPlatforms][]
+
+[`debugDefaultTargetPlatformOverride`]: {{site.api}}/flutter/foundation/debugDefaultTargetPlatformOverride.html
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/target-platform-linux-windows.md]: We prefer these links to be alphabetical, but that's just a minor nit. I wouldn't hold up merging for that.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 1 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: Done.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 2 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: OK, done.  Can the metadata lines be wrapped, or do they have to be one per line?

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 3 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: OK, fixed.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 4 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: Done.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 5 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: Probably?  I reworded it here and below to not use "shim" anyhow.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 6 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: Fixed.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 7 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: Fixed.  Thanks for the guidance.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 9 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: No, not yet.  The PR is not yet committed.  Once it is, I'll have a release number.

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 10 above)
Comment (gspencergoog) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: The *are* alphabetical, it's just that the backtick character sorts before letters. :-)

Re-sorted, ignoring formatting marks.

**Comment by gspencergoog** [comment]
@sfshaza2, OK, I've added the specific version now that the PR has landed. Is there anything else this needs?

File: src/docs/release/breaking-changes/target-platform-linux-windows.md (hunk 2 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/target-platform-linux-windows.md]: Just seeing this question now! The `description:` line in the YML meta data needs to be a single line or Travis fails.

**Pull request merged**